### ISSUE TITLE
MVC Model

### ISSUE THREAD

**Issue by Lin87** [mwolfensteinuwex/courseDevCalendar#3]
This is something that I recommend. You do not have to implement it. Just something to consider when you feel adventurous or ready to take JavaScript to the next level. It will also make your life easier when maintaining the web application.

This project is great for the model-view-controller (MVC) model. With MVC, you separate the three main components of your app/website apart for easier management and independency. So, for instance, when you have new data or need to edit data, you do not have to open the HTML file and edit the structure or add another block of HTML. You just put or edit the data in the _model_. When you ran into a situation that requires you to edit the HTML structures, you do not have to edit the all of the repeated structures. You just need to update the _view_ and/or update the _controller_ if necessary.

Model - the data coming from a database, a JSON file, an XML file, etc. _I highly recommend JSON._
View - the HTML structure template
Controller - the JS

MVC can be achieved with template binding in JS. There are many JS template binding engines out there, but I recommend that you start with [jQuery Template](https://github.com/codepb/jquery-template). There is a course on lynda.com called [JavaScript Templating](http://www.lynda.com/Web-Interaction-Design-tutorials/JavaScript-Templating/156166-2.html) to help you get started. UW employee can access lynda.com for free by following these [instructions](https://media.uwex.edu/resources/documentation/lynda/).

**Comment by mwolfensteinuwex** [comment]
Thanks Ethan! That's definitely something that occurred to me (separating content from structure), although I didn't have the language for it. I'm mostly using this project as a learning opportunity to extend my web dev knowledge so that when I collaborate with you and other developers I have a better understanding of what's what.

When I have some time I'll look into JSON if you think it's a good way to get my feet wet with databases.

**Comment by Lin87** [comment]
Yes, getting to know JSON is recommended. When developers query the database, they don't work with the raw data. They ask the database to return the data in JSON format and then go off from there.
